### ISSUE TITLE
the libblockdev package requires python2

### ISSUE THREAD

**Issue by dashea** [rhinstaller/libblockdev#23]
And I guess that's a problem in the bright new tomorrow. I guess the python2 and python3 gi/overrides files need to go in separate packages. I don't know what all that will do to packages that depend on libblockdev, or if one of the overrides should stay in the main libblockdev package, or what. Unsurprisingly there's no guidance from the Fedora guidelines for things like this.

**Comment by dashea** [comment]
Another idea: maybe the leave the files as-is and override the requires to not require python? Anyone using the python bindings is going to require python from their end anyway, and it seems nicer to just have all the gobject typelib and overrides and everything all in one package instead of creating one or two more packages for a single file

**Comment by vpodzime** [comment]
*Gtk*, *GLib*, *Pango* and some other overrides are included in the ``pygobject3-base``/``python3-gobject`` packages which seems weird to me because either these two packages require *Gtk*, *GLib*, etc. or they provide overrides for something that may not be installed.

Providing subpackages with just the overrides looks like an overkill to me. So I hope just dropping the dependencies will work.

**Comment by vpodzime** [comment]
Looking at the spec file now, I only see ``BuildRequires`` for ``python-devel`` and ``python3-devel``. So the problem is that *RPM* adds the requirements on its own.

```
$ rpm -qp --requires ~/rpmbuild/RPMS/x86_64/libblockdev-1.0-1.fc21.x86_64.rpm 
/sbin/ldconfig
/sbin/ldconfig
libbd_utils.so.0()(64bit)
libc.so.6()(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
libglib-2.0.so.0()(64bit)
python(abi) = 2.7
python(abi) = 3.4
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rtld(GNU_HASH)
```

**Comment by dashea** [comment]
Right, I was thinking the spec file could add something like this:

```
diff --git a/dist/libblockdev.spec.in b/dist/libblockdev.spec.in
index f1f4e37..bb8bfcb 100644
--- a/dist/libblockdev.spec.in
+++ b/dist/libblockdev.spec.in
@@ -25,6 +25,13 @@ BuildRequires: kmod-devel
 BuildRequires: s390utils-devel
 %endif
 
+# Do check the python directories for automatic Requires
+# The python gi.overrides files automatically add python(abi) entries to
+# the rpm Requires, but this is backwards. Anything using libblockdev from
+# python will require python, but libblockdev will work just fine without
+# python.
+%global __requires_exclude_from ^(%{python2_sitearch}|%{python3_sitearch})/gi/overrides/.*$
+
 
 %description
 The libblockdev is a C library with GObject introspection support that can be

```

**Comment by vpodzime** [comment]
Oh, I had no idea anything like that is possible. I now verified the above works!

**Comment by vpodzime** [comment]
Fixed by PR #25
